cl-gopher.lisp

```diff
@@ -478,16 +478,23 @@
             (if (and (>= (length uri) 9) (equal "gopher://" (subseq uri 0 9)))
                 (quri:uri uri)
                 (quri:uri (format nil "gopher://~a" uri))))
-           (path (quri:uri-path uri))
+           (path (quri:url-decode (quri:uri-path uri)))
            (item-type (compute-item-type uri path))
            (selector (compute-selector uri path))
+           (tab-split-selector (uiop:split-string selector :separator '(#\Tab)))
+           (selector (first tab-split-selector))
+           (terms (second tab-split-selector))
+           (gopher+string (third tab-split-selector))
            (host (quri:uri-host uri))
            (port (or (quri:uri-port uri) 70)))
-      (make-instance (class-for-type item-type)
-                     :display-string display-string
-                     :selector selector
-                     :hostname host
-                     :port port))))
+      (declare (ignore gopher+string))
+      (apply #'make-instance (class-for-type item-type)
+             :display-string display-string
+             :selector selector
+             :hostname host
+             :port port
+             (when (and terms (eq item-type :search-line))
+               `(:terms ,terms))))))
```

Comment:
For consistency you should add a `terms` slot to the `gopher-line` [class definition](https://github.com/knusbaum/cl-gopher/blob/29fbe52e3f377e24e8bde77b00789a1e32a75b90/cl-gopher.lisp#L72-L75)

Reply:
Done!